### PR TITLE
Release Management - nodejs

### PR DESCRIPTION
**Pull request by HunaidHanfee-MSFT**

### CODE REVIEW

File: samples/release-management/nodejs/README.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+---
+page_type: sample
+description: This is a sample app with capability to send notfication when user creates workitem in [Azure DevOps](https://dev.azure.com) via webhooks.
```
Comment (v-chetsh) [inline samples/release-management/nodejs/README.md:3]: notfication  => notification

File: samples/release-management/nodejs/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+- office-365
+languages:
+- nodejs
+- javascript
+extensions:
+contentType: samples
+createdDate: "28-04-2022 00:15:15"
+---
+
+# Bot to created group chat and send task notfication using Azure webhooks.
```
Comment (v-chetsh) [inline samples/release-management/nodejs/README.md:16]: notfication  => notification

Comment (v-chetsh) [inline samples/release-management/nodejs/README.md:16]: create a group chat

File: samples/release-management/nodejs/README.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+This is a sample application which demonstarates how to create a webhook on [Azure DevOps](https://dev.azure.com) and connect with Teams bot that creates a group chat and send workitems details.
```
Comment (v-chetsh) [inline samples/release-management/nodejs/README.md:18]: demonstrates

File: samples/release-management/nodejs/README.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+### 1. Start ngrok on localhost:3978
+- Open ngrok and run command `ngrok http -host-header=rewrite 3978` 
+- Once started you should see URL  `https://41ed-abcd-e125.ngrok.io`. Copy it, this is your baseUrl that will used as endpoint for Azure bot and webhook.
+
+![Ngrok](Images/NgrokScreenshot.PNG)
+
+### 2. Setup Azure DevOps service hook.
+- Follow this document- [Create Webhooks](https://docs.microsoft.com/en-us/azure/devops/service-hooks/services/webhooks?view=azure-devops) to service hook. 
+- Make sure to select trigger as *Work item created*
+- Make sure to add URL as https://{baseUrl}/api/workItem. It will look somethihng as https://41ed-abcd-e125.ngrok.io/api/workItem. *Here baseUrl is reffered to URL we get in step 1*.
```
Comment (v-chetsh) [inline samples/release-management/nodejs/README.md:48]: referred

File: samples/release-management/nodejs/teamsAppManifest/manifest.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "color": "color.png",
+    "outline": "outline.png"
+  },
+  "name": {
+    "short": "Release Management",
+    "full": "Release Management"
+  },
+  "description": {
+    "short": "TODO",
+    "full": "TODO"
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/release-management/nodejs/teamsAppManifest/manifest.json:23]: Need more details

File: samples/release-management/nodejs/helpers/devOpsHelper.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        var validEmailList = [];
+        emailList.forEach (email => {
+            var regexp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+            var isValidEmail = regexp.test(String(email.trim()).toLowerCase());
+            if (isValidEmail)
+            {
+                validEmailList.push(email.trim());
+            }
+        });
+        return validEmailList;
```
Comment (v-chetsh) [inline samples/release-management/nodejs/helpers/devOpsHelper.js:58]: One line space before it.

File: samples/release-management/nodejs/helpers/devOpsHelper.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    static GetEmailOrName(formattedString, isName)
+    {
+        var result = ""; 
+        var splitString = formattedString.split('<');
+        if (splitString.length > 0)
+        {
+            result = isName ? splitString[0] : splitString[1].replace('>','');
+        }
+        return result.trim();
```
Comment (v-chetsh) [inline samples/release-management/nodejs/helpers/devOpsHelper.js:75]: One line space before it.

File: samples/release-management/nodejs/helpers/graphHelper.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        return new Promise(async (resolve) => {
+            const config = {
+                method: 'post',
+                url: 'https://login.microsoftonline.com/' + process.env.MicrosoftAppTenantId + '/oauth2/v2.0/token',
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded'
+                },
+                data: data
+            };
+            await axios(config)
```
Comment (v-chetsh) [inline samples/release-management/nodejs/helpers/graphHelper.js:38]: One line space before it.

File: samples/release-management/nodejs/helpers/graphHelper.js [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * Gets the application internal Id.
+     * @returns Application internal Id.
+     */
+    async GetApplicationInternalId() {
+        try
+        {
+            var apps =  await this.graphClient.api(`/appCatalogs/teamsApps`).filter(`externalId eq '${process.env.MicrosoftAppId}'`).get();
+            
+            return apps.value[0].id;
```
Comment (v-chetsh) [inline samples/release-management/nodejs/helpers/graphHelper.js:131]: What if this array is null or empty.

**Review by v-chetsh — COMMENTED** [review]
Reviewed and provided my comments.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**